packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,307 @@
+  `,
+  [Variant.Light]: css`
+    background-color: ${uiColors.gray.light3};
+    border: 1px solid ${uiColors.gray.light2};
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+};
+
+interface PopoverFunctionParameters {
+  alignment: Align;
+  justification: Justification | Justify;
+  referenceElPos: ElementPosition;
+}
+
+interface TooltipProps
+  extends Omit<PopoverProps, 'active' | 'spacing' | 'refEl' | 'usePortal'> {
+  /**
+   * A slot for the element used to trigger the `Tooltip`.
+   * default: hover
+   */
+  trigger: React.ReactElement | Function;
+
+  /**
+   * Determines if a `hover` or `click` event will trigger the opening of a `Tooltip`.
+   */
+  triggerEvent?: TriggerEvent;
+
+  /**
+   * Controls component and determines the open state of the `Tooltip`
+   * default: `false`
+   */
+  open?: boolean;
+
+  /**
+   * Callback to change the open state of the `Tooltip`.
+   */
+  setOpen?: (
+    open: boolean,
+  ) => void | React.Dispatch<React.SetStateAction<boolean>>;
+
+  /**
+   * Whether the `Tooltip` will be `light` or `dark`.
+   * default: light
+   */
+  variant?: Variant;
+
+  /**
+   * id given to `Tooltip` content.
+   */
+  id?: string;
+
+  /**
+   * Callback to determine whether or not `Tooltip` should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+}
+
+/**
+ * # Tooltip
+ *
+ * Tooltip component
+ *
+ * ```
+<Tooltip
+  align='top'
+  justify='start'
+  trigger={<button>trigger</button>}
+  triggerEvent='hover'
+  variant='light'
+>
+  I am an uncontrolled Tooltip!
+</Tooltip>
+```
+ * @param props.children Content to appear inside of Tooltip.
+ * @param props.open Boolean to describe whether or not Tooltip is open.
+ * @param props.setOpen Callback to change the open state of the Tooltip.
+ * @param props.variant Whether the Tooltip should be `dark` or `light`.
+ * @param props.className Classname applied to Tooltip.
+ * @param props.align Alignment of Tooltip relative to trigger: `top`, `bottom`, `left`, `right`.
+ * @param props.justify Justification of Tooltip relative to trigger: `start`, `middle`, `end`.
+ * @param props.trigger Trigger element can be ReactNode or function.
+ * @param props.triggerEvent Whether the Tooltip should be triggered by a `click` or `hover`.
+ * @param props.id id given to Tooltip content.
+ */
+function Tooltip({
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
+  className,
+  children,
+  trigger,
+  variant = Variant.Light,
+  triggerEvent = TriggerEvent.Hover,
+  align = 'top',
+  justify = 'start',
+  id,
+  shouldClose,
+  ...rest
+}: TooltipProps) {
+  const isControlled = typeof controlledOpen === 'boolean';
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
+  const open = isControlled ? controlledOpen : uncontrolledOpen;
+  // typescript is not recognizing isControlled checks that controlledSetOpen exists
+  const setOpen = isControlled
+    ? (controlledSetOpen as Function)
```

Comment:
WOAH

Reply:
💥

Reply:
👍 Changed the logic per a previous comment, but still using the TooltipProps

Reply:
But Typescript doesn't like using TooltipProps as a value

Reply:
? What do you mean? I saw some usages of the resulting variable that TS didn't like as we use it as `setState(value)` and `setState((val) => newVal)` that we didn't define.

Reply:
`const isControlled = typeof controlledOpen === TooltipProps['open']`

Reply:
Doing it this way now per a discussion we had (perhaps in person, or further up in the thread)

Reply:
But TS throwing: `TooltipProps only refers to a type, but is being used as a value here`

Reply:
Uh I meant to only replace that one line. Not the entire expression.

Reply:
Sorry was looking at the wrong thing. I see now.

Reply:
Because in the type definition of setOpen could be undefined, TS is complaining in all usages that it can't invoke an object that is possibly undefined.

Reply:
This works:

```js
  const setOpen = isControlled && controlledSetOpen
    ? controlledSetOpen
    : uncontrolledSetOpen;
```